tests/unit_tests/layout/test_layout_array.py

```diff
@@ -222,61 +264,98 @@ def test_converting_center_coordinates(layout_array_north_four_LST_instance):
 
+    assert layout._array_center.get_altitude().value == pytest.approx(2162.35)
 
-def test_altitude_from_corsika_z(layout_array_north_four_LST_instance):
 
-    layout = layout_array_north_four_LST_instance
+def test_get_corsika_input_list(
+    layout_array_north_four_LST_instance, layout_array_south_four_LST_instance
+):
+    def one_site(layout):
+
+        layout.add_telescope(
+            telescope_name="LST-01",
+            crs_name="corsika",
+            xx=57.5 * u.m,
+            yy=57.5 * u.m,
+            tel_corsika_z=0 * u.m,
+        )
+        corsika_input_list = layout.get_corsika_input_list()
 
-    layout.add_telescope(
-        telescope_name="LST-01",
-        crs_name="corsika",
-        xx=57.5 * u.m,
-        yy=57.5 * u.m,
-        tel_corsika_z=0 * u.m,
-    )
+        assert (
+            corsika_input_list
+            == "TELESCOPE\t 57.500E2\t 57.500E2\t 0.000E2\t 12.500E2\t # LST-01\n"
+        )
 
-    assert layout._altitude_from_corsika_z(5.0 * u.m, None, "LST-01").value == pytest.approx(2147.0)
-    assert layout._altitude_from_corsika_z(None, 8848.0 * u.m, "LST-01").value == pytest.approx(
-        6706.0
-    )
-    with pytest.raises(TypeError):
-        layout._altitude_from_corsika_z(5.0, None, "LST-01")
+    one_site(layout_array_north_four_LST_instance)
+    one_site(layout_array_south_four_LST_instance)
 
 
-def test_include_radius_into_telescope_table(layout_array_north_instance, telescope_test_file):
-    telescope_table = layout_array_north_instance.read_telescope_list_file(telescope_test_file)
-    telescope_table_with_radius = layout_array_north_instance.include_radius_into_telescope_table(
-        telescope_table
-    )
-    values_from_file = [20.190000534057617, -352.4599914550781, 62.29999923706055, 9.6]
-    keys = ["pos_x", "pos_y", "pos_z", "radius"]
-    mst_10_index = telescope_table_with_radius["telescope_name"] == "MST-10"
-    for key, value_manual in zip(keys, values_from_file):
-        assert (
-            pytest.approx(telescope_table_with_radius[mst_10_index][key].value[0], 1e-2)
-            == value_manual
+def test_altitude_from_corsika_z(
+    layout_array_north_four_LST_instance, layout_array_south_four_LST_instance
+):
+    def one_site(instance, result1, result2):
+
+        instance.add_telescope(
+            telescope_name="LST-01",
+            crs_name="corsika",
+            xx=57.5 * u.m,
+            yy=57.5 * u.m,
+            tel_corsika_z=0 * u.m,
         )
 
+        assert instance._altitude_from_corsika_z(
+            pos_z=5.0 * u.m, altitude=None, tel_name="LST-01"
+        ).value == pytest.approx(result1)
+        assert instance._altitude_from_corsika_z(
+            pos_z=None, altitude=8848.0 * u.m, tel_name="LST-01"
+        ).value == pytest.approx(result2)
+        with pytest.raises(TypeError):
+            instance._altitude_from_corsika_z(5.0, None, "LST-01")
+
+    one_site(layout_array_north_four_LST_instance, 2147.0, 6706.0)
```

Comment:
Why such crazy high altitude?

Reply:
same as above. if you prefer I can change the number, but I believe it is not important at all. Now thinking about our discussions on the z - r > 0, do you think we have to adapt the code for that?

Reply:
Same answer as above. No need to adapt the code for the z - r > 0 case, we won't have it in the future.